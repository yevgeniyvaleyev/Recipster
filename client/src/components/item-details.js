import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {Row, Col} from 'react-flexbox-grid';
import styled from 'styled-components';
import { Card, Paper, CardText, CardTitle } from 'material-ui';

const Media = styled(Paper)`
  max-height: 400px;
  min-height: 400px;
  background-repeat: no-repeat;
  background-size: cover;
`;

export class ItemDetails extends Component {
  
  render () {
    const { item, loading } = this.props;

    return (
      <Card>
        <CardTitle title={item.title} subtitle={item.publisher} />
        <CardText>
          <Row>
            <Col lg={4} md={6} sm={12}>
              <Media style={{backgroundImage: `url(${item.image_url})`}} alt={item.title + ', image'} />
            </Col>
            <Col lg={4} md={6} sm={12}>
              <h3>Ingredients</h3>
              {(!loading && item.ingredients)
                ? <ol>
                  {item.ingredients.map((ingredient, index) => 
                    <li key={index}>{ingredient}</li>
                  )}
                </ol>
                : <div>Ingredients are being loaded...</div>
              }

            </Col>
            <Col lg={4} md={6} sm={12}>
              <h3>Info</h3>
              <ul>
                <li>Publisher: <a href={item.publisher_url}>{item.publisher}</a></li>
                <li>Social rank: {item.social_rank}</li>
                <li>ID of recipe: {item.recipe_id}</li>
                <li>Recipe source: <a href={item.source_url}>link</a></li>
              </ul>
            </Col>
          </Row>
        </CardText>
      </Card>
    )
  }
}

ItemDetails.propTypes = {
  item: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired
};