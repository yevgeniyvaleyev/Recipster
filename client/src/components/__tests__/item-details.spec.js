import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { ItemDetails } from '../item-details';
import { BrowserRouter } from 'react-router-dom';
import { Grid } from 'react-flexbox-grid';

test('should render component while loading ingredients', () => {
  const loading = true;
  const item = {
    title: 'test title',
    image_url: 'test.png',
    publisher: 'test',
    recipe_id: 1
  }
  const component = renderer.create(
    <MuiThemeProvider>
      <BrowserRouter>
        <Grid>
          <ItemDetails item={item} loading={loading} />
        </Grid>
      </BrowserRouter>
    </MuiThemeProvider>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('should render component when loaded ingredients', () => {
  const loading = false;
  const item = {
    title: 'test title',
    image_url: 'test.png',
    ingredients: ['ingredient'],
    publisher: 'test',
    recipe_id: 1
  }
  const component = renderer.create(
    <MuiThemeProvider>
      <BrowserRouter>
        <Grid>
          <ItemDetails item={item} loading={loading} />
        </Grid>
      </BrowserRouter>
    </MuiThemeProvider>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

