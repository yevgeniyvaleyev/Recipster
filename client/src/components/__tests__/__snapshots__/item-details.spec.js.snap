// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component when loaded ingredients 1`] = `
<div
  className="container"
>
  <div
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "zIndex": 1,
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": 8,
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
            "padding": 16,
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 24,
              "lineHeight": "36px",
              "muiPrepared": true,
            }
          }
        >
          test title
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          test
        </span>
      </div>
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 14,
            "muiPrepared": true,
            "padding": 16,
          }
        }
      >
        <div
          className="row"
        >
          <div
            className="col-lg-4 col-md-6 col-xs-12"
          >
            <div
              alt="test title, image"
              className="sc-bdVaJa egwyhe"
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "backgroundColor": "#ffffff",
                  "backgroundImage": "url(test.png)",
                  "borderRadius": 2,
                  "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                         0 1px 4px rgba(0, 0, 0, 0.12)",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "Roboto, sans-serif",
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            />
          </div>
          <div
            className="col-lg-4 col-md-6 col-xs-12"
          >
            <h3>
              Ingredients
            </h3>
            <ol>
              <li>
                ingredient
              </li>
            </ol>
          </div>
          <div
            className="col-lg-4 col-md-6 col-xs-12"
          >
            <h3>
              Info
            </h3>
            <ul>
              <li>
                Publisher: 
                <a
                  href={undefined}
                >
                  test
                </a>
              </li>
              <li>
                Social rank: 
              </li>
              <li>
                ID of recipe: 
                1
              </li>
              <li>
                Recipe source: 
                <a
                  href={undefined}
                >
                  link
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render component while loading ingredients 1`] = `
<div
  className="container"
>
  <div
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "zIndex": 1,
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": 8,
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
            "padding": 16,
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 24,
              "lineHeight": "36px",
              "muiPrepared": true,
            }
          }
        >
          test title
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          test
        </span>
      </div>
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 14,
            "muiPrepared": true,
            "padding": 16,
          }
        }
      >
        <div
          className="row"
        >
          <div
            className="col-lg-4 col-md-6 col-xs-12"
          >
            <div
              alt="test title, image"
              className="sc-bdVaJa egwyhe"
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "backgroundColor": "#ffffff",
                  "backgroundImage": "url(test.png)",
                  "borderRadius": 2,
                  "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                         0 1px 4px rgba(0, 0, 0, 0.12)",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "Roboto, sans-serif",
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            />
          </div>
          <div
            className="col-lg-4 col-md-6 col-xs-12"
          >
            <h3>
              Ingredients
            </h3>
            <div>
              Ingredients are being loaded...
            </div>
          </div>
          <div
            className="col-lg-4 col-md-6 col-xs-12"
          >
            <h3>
              Info
            </h3>
            <ul>
              <li>
                Publisher: 
                <a
                  href={undefined}
                >
                  test
                </a>
              </li>
              <li>
                Social rank: 
              </li>
              <li>
                ID of recipe: 
                1
              </li>
              <li>
                Recipe source: 
                <a
                  href={undefined}
                >
                  link
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
